lightningd/jsonrpc.c

```diff
@@ -884,3 +885,71 @@ bool json_tok_wtx(struct wallet_tx * tx, const char * buffer,
 	}
         return true;
 }
+
+static bool json_tok_command(struct command *cmd, const char *name,
+			     const char *buffer, const jsmntok_t *tok,
+			     const jsmntok_t **out)
+{
+	cmd->json_cmd = find_cmd(buffer, tok);
+	if (cmd->json_cmd)
+		return (*out = tok);
+
+	command_fail(cmd, JSONRPC2_INVALID_PARAMS,
+		     "'%s' of '%.*s' is invalid",
+		     name, tok->end - tok->start, buffer + tok->start);
+	return false;
+}
+
+static void json_check(struct command *cmd,
+		       const char *buffer, const jsmntok_t *params)
+{
+	jsmntok_t *mod_params;
+	const jsmntok_t *name_tok;
+	bool ok;
+	struct json_stream *response;
+
+	if (cmd->mode == CMD_USAGE) {
+		mod_params = NULL;
+	} else {
+		mod_params = json_tok_copy(cmd, params);
+		cmd->allow_unused = true;
+	}
+
+	if (!param(cmd, buffer, mod_params,
+		   p_req("command_to_check", json_tok_command, &name_tok),
```

Comment:
I would remove "_to_check" here, since it makes the JSON API neater (the context is pretty clear).

Reply:
... or just call it 'check'?

Reply:
`command` seems better to me.

Reply:
I discussed my reason for choosing such an awkward name in the commit message for https://github.com/ElementsProject/lightning/pull/2123/commits/e45dd7f7c8170526c177a247700979442c599cd6. Basically, it has to be unique to all other parameter names for all commands. But I'm fine with changing it to whatever you are comfortable with. I can also add a developer mode check to make sure it is not used again. Note @ZmnSCPxj, we already are using `command` for the help command.

Reply:
Note that you *can't* make it unique, since 'check check' is supposed to work :)  But clashing with some other command would be painful, so happy to stop bikeshedding now :)